examples/matrix_multiplier/tests/test_matrix_multiplier.py

```diff
@@ -0,0 +1,143 @@
+                b_matrix = self.bus.B.value
+
+                # Calculate the expected result of C
+                c_expected = [
+                    BinaryValue(
+                        sum(
+                            [
+                                a_matrix[(i*A_COLUMNS_B_ROWS)+n] * b_matrix[(n*B_COLUMNS)+j]
+                                for n in range(A_COLUMNS_B_ROWS)
+                            ]
+                        ),
+                        n_bits=(DATA_WIDTH * 2) + math.ceil(math.log2(A_COLUMNS_B_ROWS)),
+                        bigEndian=False
+                    )
+                    for i in range(A_ROWS)
+                    for j in range(B_COLUMNS)
+                ]
+
+                self._recv(c_expected)
+
+
+class MatrixOutMonitor(MatrixMonitor):
+    """Monitor outputs from Matrix Multiplier module.
+
+    Capture result matrix for each multiplication operation.
+    """
+    _signals = {"C": "c_o", "valid": "valid_o"}
+
+    async def _monitor_recv(self):
+        while True:
+            await RisingEdge(self.clock)
+            await ReadOnly()
+
+            if self.bus.valid.value:
+                c_actual = self.bus.C.value
+                self._recv(c_actual)
+
+
+async def test_multiply(dut, a_data, b_data):
+    """Test multiplication of many matrices."""
+
+    cocotb.fork(Clock(dut.clk_i, 5, units='ns').start())
+
+    # Configure Scoreboard to compare module results to expected
+    expected_output = []
+
+    def add_expected(transaction):
+        expected_output.append(transaction)
+
+    in_monitor = MatrixInMonitor(dut, callback=add_expected)
+    out_monitor = MatrixOutMonitor(dut)
+
+    scoreboard = Scoreboard(dut)
+    scoreboard.add_interface(out_monitor, expected_output)
+
+    # Initial values
+    dut.valid_i <= 0
+    dut.a_i <= next(gen_a(1, lambda x: 0))
+    dut.b_i <= next(gen_b(1, lambda x: 0))
+
+    # Reset DUT
+    dut.reset_i <= 1
+    for _ in range(3):
+        await RisingEdge(dut.clk_i)
+    dut.reset_i <= 0
+
+    # Do multiplication operations
+    for A, B in zip(a_data(), b_data()):
+        await RisingEdge(dut.clk_i)
+        dut.a_i <= A
+        dut.b_i <= B
+        dut.valid_i <= 1
+
+        await RisingEdge(dut.clk_i)
+        dut.valid_i <= 0
+
+    await RisingEdge(dut.clk_i)
+
+    raise scoreboard.result
+
+
+def gen_matrix(func, rows, cols):
+    return [func(DATA_WIDTH) for row in range(rows) for col in range(cols)]
+
+
+def gen_a(num_samples=NUM_SAMPLES, func=getrandbits):
+    """Generate random matrix data for A"""
+    for i in range(num_samples):
+        yield gen_matrix(func, A_ROWS, A_COLUMNS_B_ROWS)
+
+
+def gen_b(num_samples=NUM_SAMPLES, func=getrandbits):
+    """Generate random matrix data for B"""
+    for i in range(num_samples):
+        yield gen_matrix(func, A_COLUMNS_B_ROWS, B_COLUMNS)
+
+
+factory = TestFactory(test_multiply)
+factory.add_option('a_data', [gen_a])
+factory.add_option('b_data', [gen_b])
```

Comment:
```suggestion
factory.add_option('a_data', [lambda: (gen_a() for _ in range(NUM_SAMPLES)])
factory.add_option('b_data', [lambda: (gen_b() for _ in range(NUM_SAMPLES)])
```

Reply:
The problem is lambdas mess up the auto-generated test description:
```
Description: Automatically generated test

    a_data: <lambda> (No docstring supplied)
    b_data: <lambda> (No docstring supplied)
```

Reply:
Fun. Oh well.